### PR TITLE
Feature/add log clickthrough

### PR DESCRIPTION
**Pull request by damienwhaley**
Added the log_clickthrough functionality to record the search result that the user clicked on. This helps provide feedback to the engine to improve the search results.

Updated fixtures with new data from testing instance.

Updated dependencies in package.json to latest versions.

Made small changes in README documentation. Specifically changed "Autocomple" to "Suggest" to reflect the name of the function call. Also added documentation for "Click" function.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,9 +42,9 @@ Search for `cats` on the engine `my-engine` with filters and facets:
       console.log(res)
     })
 
-### Autocomplete
+### Suggest
```
Comment (goodroot) [inline README.md]: Thanks for the PR! We'd like to keep this as **Autocomplete**. The reason is that Autocomplete is consistent with the [documentation](**https://swiftype.com/documentation/site-search/autocomplete**), through from the product and out to marketing. 👍

File: lib/analytics.js [hunk 2]
```diff
@@ -36,6 +36,10 @@ Analytics.prototype = {
 
   topNoResultQueries: function(params, callback) {
     this.client.get('/engines/{engine}/analytics/top_no_result_queries.json', params, callback)
+  },
+
+  logClickthrough: function(params, callback) {
+    this.client.post('/engines/{engine}/document_types/{documentType}/analytics/log_clickthrough.json', params, callback);
```
Comment (JasonStoltz) [inline lib/analytics.js]: The convention in this project is no semi-colons.

File: test/swiftypeTest.js [hunk 3]
```diff
@@ -35,7 +35,23 @@ describe('swiftype', function() {
...
+    var client = new Swiftype({ apiKey: apiKey })
+
+    it('logs the click event', function(done) {
+      client.click({
+        engine: myEngine,
+        q: 'awesome',
+        id: '5b886b63827a6656794c6eaf',
+        documentType: 'page'
+      }, function(err, res) {
+        assert.equal(null, err)
```
Comment (JasonStoltz) [inline test/swiftypeTest.js]: Could we test for a 200 response here?

File: test/analyticsTest.js [hunk 4]
```diff
@@ -177,8 +177,20 @@ describe('analytics', function() {
...
\ No newline at end of file
+
+  it('logs the click event', function(done) {
+    client.click({
+      engine: myEngine,
+      q: 'awesome',
+      id: '5b886b63827a6656794c6eaf',
+      documentType: 'page'
+    }, function(err, res) {
+      assert.equal(null, err)
```
Comment (JasonStoltz) [inline test/analyticsTest.js]: Could we test for a 200 response here?

File: README.md (hunk 1 above)
Comment (damienwhaley) [reply README.md]: No problems - I'll switch this back.

File: lib/analytics.js (hunk 2 above)
Comment (damienwhaley) [reply lib/analytics.js]: No problems - will follow the convention

File: test/swiftypeTest.js (hunk 3 above)
Comment (damienwhaley) [reply test/swiftypeTest.js]: Good idea - will do

File: test/analyticsTest.js (hunk 4 above)
Comment (damienwhaley) [reply test/analyticsTest.js]: Good idea - will do

**Comment by damienwhaley** [comment]
I'll merge master into this branch to enable CI to run, and I'll revert the fixtures and just add the two new ones for click.

Pull request incoming soon!

**Comment by JasonStoltz** [comment]
@damienwhaley This is a bit of a challenge. This library is, and ideally will continue to be, compatible with older versions of node. For that reason I'd like to run CI on older versions of node. 

At the same time, we'd like to continue to improve our test suite and use the latest tooling. The challenge with that, is that as we upgrade our test tooling, it slowly becomes incompatible with older versions of node, which limits our ability to run CI against those older node versions.

I suppose it'd be fine to slowly remove support for those older versions of node. The question in my mind is then, do we rev the client library to a new major version and remove support officially?

In the meantime, I think you're good to go ahead and remove node 4 from the travis configuration. I'd say also bump the node engine version in package.json file to be >= v6.0 to reflect that we're no longer actively testing for versions less than 6.

I'll consider what to do about versioning in the meantime.

**Comment by JasonStoltz** [comment]
Actually, @damienwhaley, the simplest path forward here may be to simply revert the `replay` upgrade.

File: test/analyticsTest.js (hunk 4 above)
Comment (JasonStoltz) [reply test/analyticsTest.js]: That makes sense.

**Comment by damienwhaley** [comment]
I've downgraded replay which has allowed the tests to pass on all versions in Travis

**Comment by JasonStoltz** [comment]
@damienwhaley version 0.4.0 has been release, and includes these changes: https://www.npmjs.com/package/swiftype. Thank you again for your contribution.

**Pull request merged**